modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceConfigurationManager.java

```diff
@@ -257,40 +258,111 @@ private void validate(ServiceTemplateEntity serviceTemplateEntity,
                 serviceConfigurationUpdateStorage.listServiceConfigurationUpdateRequests(model);
         if (CollectionUtils.isEmpty(requests)) {
-            return ResponseEntity.status(HttpStatus.NO_CONTENT)
-                    .body(new ServiceConfigurationChangeRequest());
+            return null;
         }
-        Optional<ServiceConfigurationUpdateRequest> latestRequest = requests.stream()
+        Optional<ServiceConfigurationUpdateRequest> oldestRequestOptional = requests.stream()
+                .filter(request -> Objects.isNull(request.getResourceName())
+                        || resourceName.equals(request.getResourceName()))
                 .filter(request -> request.getServiceOrderEntity() != null
                         && request.getServiceOrderEntity().getStartedTime() != null)
-                .sorted(Comparator.comparing(request ->
-                        request.getServiceOrderEntity().getStartedTime())).findFirst();
+                .min(Comparator.comparing(request ->
+                        request.getServiceOrderEntity().getStartedTime()));
+        return oldestRequestOptional.orElse(null);
+    }
+
+    private void validateConfigManager(String serviceId, String configManager,
+                                       String resourceName, List<DeployResource> deployResources) {
+        Map<String, List<DeployResource>> groupNameMap =
+                deployResources.stream()
+                        .collect(Collectors.groupingBy(DeployResource::getGroupName));
+        Map<String, List<DeployResource>> resourceNameMap =
+                deployResources.stream()
+                        .collect(Collectors.groupingBy(DeployResource::getResourceName));
+        if (!groupNameMap.containsKey(configManager)) {
+            String errorMsg = String.format("The service with serviceId %s does not contain "
+                    + "a group with group_name %s", serviceId, configManager);
+            log.error(errorMsg);
+            throw new IllegalArgumentException(errorMsg);
 
-        if (latestRequest.isEmpty()) {
-            return  ResponseEntity.status(HttpStatus.NO_CONTENT)
-                    .body(new ServiceConfigurationChangeRequest());
         }
-        ServiceConfigurationUpdateRequest request = latestRequest.get();
-        request.setStatus(ServiceConfigurationStatus.PROCESSING);
-        serviceConfigurationUpdateStorage.storeAndFlush(request);
-        ServiceConfigurationChangeRequest serviceConfigurationChangeRequest
-                = getChangeRequestForAgent(request);
-        return ResponseEntity.status(HttpStatus.OK).body(serviceConfigurationChangeRequest);
+        if (!resourceNameMap.containsKey(resourceName)) {
+            String errorMsg = String.format("The service with serviceId %s does not contain "
+                    + "a resource with resource_name %s", serviceId, resourceName);
+            log.error(errorMsg);
+            throw new IllegalArgumentException(errorMsg);
```

Comment:
would prefer to send any of our custom exception. If we do not have any thing that fits, we must create a new one.